apps/dashboard/app/javascript/dynamic_forms.js

```diff
@@ -158,16 +158,29 @@ function makeChangeHandlers(prefix){
             });
           }
       });
+    } else if(element['type'] == "checkbox") {
+      let data = $(element).data();
+      let keys = Object.keys(data);
+      if(keys.length !== 0) {
+        keys.forEach((key) => {
+          if(key.startsWith('hide')) {
+            let tokens = parseCheckedWhen(key);
+            if(tokens !== undefined){
+              addHideHandler(element['id'], tokens['value'], tokens['key'], data[key]);
+            }
+          }
+        });
+      }
     }
   });
 
   initializing = false;
 };
 
-function addHideHandler(optionId, option, key,  configValue) {
+function addHideHandler(optionId, option, key, configValue) {
   const changeId = idFromToken(key.replace(/^hide/,''));
 
-  if(hideLookup[optionId] === undefined) hideLookup[optionId] = new Table(changeId, 'option_value');
+  if(hideLookup[optionId] === undefined) hideLookup[optionId] = new Table(changeId, undefined);
```

Comment:
I think having `'option_value'` here was a mistake and/or worked by coincidence. I'm fairly sure this table should only have 1 dimension.